Explorer/Assets/DCL/PerformanceAndDiagnostics/Optimization/Pools/GameObjectPool.cs

```diff
@@ -13,19 +13,22 @@ public class GameObjectPool<T> : IComponentPool<T> where T: Component
 
         private readonly IExtendedObjectPool<T> gameObjectPool;
         private readonly Transform parentContainer;
-        private readonly Transform rootContainer;
 
-        private readonly Action<T> onRelease;
-        private readonly Action<T> onGet;
+        private readonly Action<T>? onRelease;
+        private readonly Action<T>? onGet;
 
         public int CountInactive => gameObjectPool.CountInactive;
 
-        public GameObjectPool(Transform rootContainer, Func<T> creationHandler = null, Action<T> onRelease = null, int maxSize = 2048, Action<T> onGet = null)
+        public Transform PoolContainerTransform => parentContainer;
+
+        public GameObjectPool(Transform rootContainer, Func<T>? creationHandler = null, Action<T>? onRelease = null, int maxSize = 2048, Action<T>? onGet = null)
         {
             parentContainer = new GameObject($"POOL_CONTAINER_{typeof(T).Name}").transform;
+#if UNITY_EDITOR
             parentContainer.SetParent(rootContainer);
-            this.onRelease += onRelease;
-            this.onGet += onGet;
+#endif
+            if (onRelease != null) this.onRelease += onRelease;
```

Comment:
should we handle this in the disposal?

Reply:
for what I see during disposal we do not unsubscribe these at all, we just remove all elements... but as these pools are not really disposed except when closing the game, it might not matter?
I will create a ticket to properly refactor this pool as right now it is not perfectly functional tbh...